core/src/main/java/org/apache/gravitino/job/JobManager.java

```diff
@@ -300,6 +300,40 @@ public boolean deleteJobTemplate(String metalake, String jobTemplateName) throws
         });
   }
 
+  @Override
+  public JobTemplateEntity alterJobTemplate(
+      String metalake, String jobTemplateName, JobTemplateChange... changes)
+      throws NoSuchJobTemplateException, IllegalArgumentException {
+    checkMetalake(NameIdentifierUtil.ofMetalake(metalake), entityStore);
+
+    NameIdentifier jobTemplateIdent = NameIdentifierUtil.ofJobTemplate(metalake, jobTemplateName);
+    return TreeLockUtils.doWithTreeLock(
+        jobTemplateIdent,
+        LockType.WRITE,
+        () -> {
+          try {
+            return entityStore.update(
+                jobTemplateIdent,
+                JobTemplateEntity.class,
+                Entity.EntityType.JOB_TEMPLATE,
+                jobTemplateEntity ->
+                    updateJobTemplateEntity(jobTemplateIdent, jobTemplateEntity, changes));
+          } catch (NoSuchEntityException e) {
+            throw new NoSuchJobTemplateException(
+                "Job template with name %s under metalake %s does not exist",
+                jobTemplateName, metalake);
+          } catch (IOException ioe) {
+            throw new RuntimeException(ioe);
+          } catch (EntityAlreadyExistsException e) {
+            throw new RuntimeException(
+                "Job template with the same name "
+                    + jobTemplateName
```

Comment:
The message here is incorrect. The name should be the value from `updateJobTemplateEntity(jobTemplateIdent, jobTemplateEntity, changes)`.

Reply:
And the message `because job template doesn't support rename` is quite misleading.